---
title: "Homework_1"
author: "Justin Rigby"
date: "20 January 2018"
output:
  html_document: default
  pdf_document: default
pandoc_args: --latex-engine=xelatex
---
  
#Review 1 
  
###R1. You flip what you think is a fair coin ten times and get heads every time.
  
(a) (1pt) What is the probability of this result if the coin’s true probability
of landing heads p = 0.5  
  
$Pr(k; n, p) = Pr(10; 10, 0.5) = \binom{10}{10}{0.5}^{10}(1-(0.5))^{10-10}=0.0009765625$  
  
  
(b) (1pt) What p would make this outcome (ten out of ten heads) have
probability 0.01  
  
$Pr(k; n, p) = Pr(10; 10, P) = \binom{10}{10}{p}^{10}(1 - p)^{10-10}=p^{10} = 0.01$
  
$p^{10} = 0.01$ => $10*ln(p) = ln (0.01)$ => $ln(p) = \frac{ln(0.01)}{10}$ => $ln(p) = -0.460517$  
  
$\therefore p = 0.6309573$  
  
(c) (1pt) What is the mean of a binomial distribution with n trials and
outcome probability p? Simplify as much as possible.  
  
  
$\mu = \sum k \binom{n}{k}p^{k}(1-p)^{n-k} = \sum k \frac{n!}{k!(n-k)!}p^{k}(1-p)^{n-k}=\sum \frac{n!p^{k}(1-p)^{n-k}}{(k-1)!(n-k)!}$  
  
###R2. Your favorite bacterium generates a Poisson-distributed number of progeny
with λ = 1.5 average number of progeny per hour.   
(a) (1pt) What is the probability that a bacterium generates zero offspring
in an hour?  
  
$Pr(k; \lambda)=Pr(0;1.5)=\frac{\lambda^{k}e^{-\lambda}}{k!}=\frac{1.5^{0}e^{-1.5}}{0!} = 0.2231302$  
  
(b) (1pt) What is the probability that it generates at least one offspring?  
 
$Pr(k; \lambda)=Pr(1;1.5)=\frac{\lambda^{k}e^{-\lambda}}{k!}=\frac{1.5^{1}e^{-1.5}}{1!} = 0.3346952$
  
(c) (1pt) What are the mean and standard deviation of the number of
offspring per hour, both in terms of λ and in actual numbers for this
bacterium?  
  
For actual numbers
  
$\sum kPr(k) = 1.5$
  
In Terms of $\lambda$  


###R3. A yeast cell buds to produce a daughter cell every two hours. Assume that,
in a population of 10 8 cells synchronized to bud simultaneously, this process
is well-described by the function n(t + 1) = Rn(t) where n(t) is the number
of cells at time t (in hours) and R is the rate of growth.  
  
  
(a) (1pt) What is R?  
R is the ratio of growth in the population to its original size.
  
$R = \frac{n(t+1)}{n(t)}$

(b) (1pt) If we now ask for the instantaneous rate of growth r rather than
the per-unit-time rate of growth, we have that dn(t)
dt = rn(t). What is
r in terms of R? Hint: write out the solution for n(t) in each case.
  
$r = \frac{dtR}{n(t+1)}$
    
  
  
###1. As discussed in class, organisms struggle for survival against statistics as well as each other. Download the expgrowth.R script from the Canvas site (Assignments/HW1/src) and run it in RStudio. Manipulate the population parameters near the top: n0, the number of starting organisms, and max.gen, the number of generations to track. (Feel free to explore further and alter the fitness and so on.)
  
  
(a) (2pts) Run 100 replicates for 100 generations with initial population
sizes of n0 = 1, 10, 100, and 1000. At each n0, after 100 generations,
what fraction of the populations have died out (i.e., have n = 0)?
  
n0 = 1      | Fraction of dead populations after 100 generations = 0.12   
n0 = 10     | Fraction of dead populations after 100 generations = 0.83  
n0 = 100    | Fraction of dead populations after 100 generations = 0.17   
n0 = 1000   | Fraction of dead populations after 100 generations = 0   
  
  
  
  
(b) (2pts) Now run 100 replicates with initial population sizes of n0 = 1,
10, 100, and 1000, but run for n0 generations (1 generation for n0 = 1,
100 generations for n0 = 100, etc.). What do you notice? 
  
n0 = 1      | Fraction of dead populations after 1 generations = 0.39  
n0 = 10     | Fraction of dead populations after 10 generations = 0.15  
n0 = 100    | Fraction of dead populations after 100 generations = 0.18  
n0 = 1000   | Fraction of dead populations after 1000 generations = 0.07  
  
As you increase the amount of generations the rate at which the allele is lost in the popualtion tends to go up -- but this is also a function of the starting size of the intial population which grows with a positive slope when plotted with the number of generations. Another way to look at the results is to remember that with every generation it is a conditional probability of retaining the allele in the next generation based upon the allele frequency in the prior generation -- which is not modified by a selection coefficent.  
  
  
###2. In $\beta$ lactamases, several sites commonly accumulate changes that alter resistance to various $\beta$-lactam antibiotics. We would like to know where these residues are relative to the antibiotic itself. Download the TEM1-imipenemfrom-1BT5.pse file from the Canvas site (Assignments/HW1/data). Do a little preparation: show the TEM-1 molecule as a gray cartoon on a white background, show the $\beta$-lactam inhibitor imipenem, and orient the view:
load your_file_path/TEM1-imipenem-from-1BT5.pse  
hide everything  
bg_color white  
show cartoon, TEM-1  
color gray70, TEM-1  
orient TEM-1  
show sticks, imipenem  
  

(a) (2pts) Select residues M69, S70, R164, M182 and G238, which have been inferred to play important roles in the evolution of drug resistance:  
select resi 69+70+164+182+238  
Show these residues as red spheres:  
show spheres, sele; color red, sele  
Raytrace the result and print/paste it into your answers:
ray; png your_file_path/tem1-residues-highlighted.png  
  
[See the folder]  
   
   
(b) (2pts) Which of these residues is closest to the substrate? Which is farthest away?
  
The residue 182 seems to be the closest to the substrate since the space around it has an opening (an area without $\beta$-sheets and $\alpha$-helixes) which would allow for an antibiotic to dock. The residue 70 is the futherest away residue relative to the implied substrate site. 
  
  
###3. Highly conserved amino acid sites are often assumed to be functionally important. Let’s use Python to study the diversity at amino acid sites in TEM-1 β-lactamases. Download the CalculateNumAaPerSite.py script and the tem1-alignment.fasta file from the Canvas site. If you haven’t done so already, read the “Program-mng and Installation” guide document we’ve written, install Python and the required modules (biopython), and organize your homework directory structure accordingly.   
###The multiple sequence alignment contains 125 TEM-1 β-lactamase se-quences ceah of which is 287 amino acids long.  
###Open up CalculateNumAaPerSite.py in Spyder. Change the base_directory variable to your homework directory’s path. Note that you need to have your directories set up correctly, as we detail in the Programming Tips guide. Read the script and comments carefully to understand what it will do. Run the script.  
  
  
a) (1pt) Examine the output histogram SiteDiversityHist.png in your hw1/results folder. How many invariant sites did you find?  
  
There are forteen invariant sites reported by the program.
  
(b) (2pts) Which sites are invariant? Modify the script to identify these invariant sites.  

66 F  
70 S  
73 K  
81 L  
107 P  
125 A  
131 D  
134 A  
136 N  
157 D  
166 E  
179 D  
199 L  
236 G  
  

(c) (1pt) Return to PyMol, as in the previous question. Show these invariant sites as blue spheres. Print/paste the result.  
  [See folder]  
  
(d) (2pts) What is the relationship between the absolutely conserved sites and those found to be functionally important in the evolution of resistance?   
  
Only one residue (number 70) is found in the invariant sites, along with the residues, and is assumed to have an impact on the evolution of drug resistance from the previous question. Otherwise, the majority of the invariant residues tend to culminate in the section of the protein where the antibiotic would reside. Thus, we can infer that the substrate area of the protein is generally conserved, but it only takes a few residues to mutate to cause a change which leads to antibiotic resitstance.  
    
   
4. Hydrolysis of β-lactam antibiotics allows bacteria to grow. Suppose we have the chemical reaction describing the spontaneous hydrolysis of ampicillin:kA −−→ ∅ with k = 0.01 s −1    
  
(a) (1pt) On average, how fast does 10 μM ampicillin degrade sponta-eously (in μM per second) at the beginning of the reaction?  
  
Use a simple rate calculation (in terms of velocity).
$v = [S]*k= 10~\mu M * 0.01~s^{-1}=0.1$
  
(b) (1pt) Plot [A](t) as a function of time in R (see tip below).  
  
```{r}
#Time in seconds
t <- 1:100
#Spontaneous hydrolysis of Ampicillin
k <- 0.1

plot(t, exp(k*t), type='l', col = 'black', 
     xlab = 'Time (s)', ylab = "Amount (uM/s)")
```
  
(c) (1pt) Suppose a β-lactamase L catalyzes this hydrolysis: when bound to β-lactamase, ampicillin hydrolyzes 10,000 times faster than its spontaneous rate. What is k cat ?
  
$K_cat =10,000 * k = 10,000 * 0.1~s^{-1}= 100 s^{-1}$

(d) (2pts)Michaelis constant K M for this reaction?  
  
$K_m = \frac{K_{off} + K_{cat}}{K_{on}} = \frac{10 s^{-1}+ 100 s^{-1}}{1,000 (\mu M)^{-1}}= 0.110 ~\frac{\mu M}{s}$
  
(e) (2pts) On average, how fast is the degradation of A at the beginning of a reaction that again starts with an initial concentration of 10 μM ampicillin, along with an effective concentration of 1 μM β-lactamase?
   
  
$v_0 = \frac{V_{max}~[S]}{K_m + [S]}= \frac{E_0 K_{cat}[S]}{K_m + [S]}= \frac{1\mu M*100 s^{-1}*10\mu M}{0.110 \frac{\mu M}{s}+10\mu M}=98.912\frac{\mu M}{s}$
    
  
  